---
title: "PHQ9GAD7WSAS h2 and rg estimates from GCTA"
author: "Megan Skelton"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    theme: flatly
    toc: true
    toc_float: 
      collapsed: true
    code_folding: hide
    df_print: paged
    highlight: monochrome
    number_sections: no
---

Heritability and genetic correlation estimates using GCTA software https://yanglab.westlake.edu.cn/software/gcta/#Overview 

# Set up
```{r setup, include = F}

knitr::opts_chunk$set(results = 'asis',     
                      comment = NA,
                      prompt = FALSE,
                      cache = FALSE,
                      warning = FALSE,
                      message = FALSE,
                      fig.height = 5, 
                      fig.width = 10,
                      fig.align = "center")

```

```{r Clear global environment, include = F}

rm(list = ls())

```

```{r Install packages}

#load functions
source(file = "../functions/package_check.R")
source(file = "../functions/sumscores.R")

formatv <- function(x, ...) {
  mapply(format, x, scientific = abs(x) < 0.001, ...)
} #scientific notation used in tables if less than stated value   

packages = c(
  "truncnorm", #simulate random nd within range
  "kableExtra", #format output
  "tidyverse" #multiple packages for tidy coding, cleaning, wrangling, visualising data
  )

package_check(packages)

```

```{r GLAD colour palette for plots}

glad_palette = c("#efc00b", "#b7dee8")
my.palette = c("#E7298A", "#009974", "#D95F02", "#7570B3") #colorbrewer Dark2 reordered

```

```{r ggplot theme}

my_theme = theme(
  axis.title.x = element_text(colour = "black", size = 13),
  axis.title.y = element_text(colour = "black", size = 13, vjust = .5),
  axis.text = element_text(colour = "black", size = 11),
  axis.ticks = element_blank(),
  panel.background = element_blank(),
  legend.text = element_text(colour="black", size=11),
  legend.key = element_blank())

```


# Valid chromosomes

GLAD_analysis_subset_maf1_sample95_snp98_hwe8 (EUR & no duplicates)  
*564,376 variants and 17,130 individuals pass filters and QC  
GCTA throws error due to chr "19_KI270938v1_alt" (this was removed in conversion to MT for regenie); remove this chr
```{bash Remove illegal chr number, eval = F}

cd ~/credentials
gendatapath=$(sed '1q;d' paths.config.sh) 
workdir=$(sed '2q;d' paths.config.sh) 
cd ${workdir}

module add apps/plink2/2.0.0a2
plink2 \
--bfile ${workdir}input/GLAD_analysis_subset_maf1_sample95_snp98_hwe8 \
--chr 1-22 \
--allow-extra-chr \
--make-bed \
--out ${workdir}input/GLAD_analysis_subset_EUR_nodup_maf1_sample95_snp98_hwe8_filtered

```
551,286 variants remaining

# Create GRM

## Make GRM
```{bash make grm, eval = F}

cd ~/credentials
workdir=$(sed '2q;d' paths.config.sh) 

rm ${workdir}scripts/make_grm.sh
cat <<'EOT'>> ${workdir}scripts/make_grm.sh
#!/bin/bash -l
#SBATCH --partition brc,shared
#SBATCH --time=0-48:00:00
#SBATCH --nodes=1
#SBATCH --ntasks=8
#SBATCH --mem-per-cpu=19G
#SBATCH --job-name=make_grm
#SBATCH --array=1-10
#SBATCH --output=../output/extraqc_make_grm_full%j.out

gendatapath=$(sed '1q;d' paths.config.sh) 
workdir=$(sed '2q;d' paths.config.sh) 
gcta=$(sed '5q;d' paths.config.sh)

${gcta} \
--bfile ${workdir}input/GLAD_analysis_subset_EUR_nodup_maf1_sample95_snp98_hwe8_filtered \
--make-grm-part 10 ${SLURM_ARRAY_TASK_ID} \
--maf 0.01 \
--out ${workdir}results/gcta/glad_gcta_full \
--thread-num 8

EOT

sbatch ${workdir}scripts/make_grm.sh

```

## Merge GRM parts
```{bash merge grm chr, eval = F}

cat glad_gcta_full.part_10*.grm.id > glad_gcta_extraqc_full.grm.id
cat glad_gcta_full.part_10*.grm.bin > glad_gcta_extraqc_full.grm.bin
cat glad_gcta_full.part_10*.grm.N.bin > glad_gcta_extraqc_full.grm.N.bin

# remove the parts files, no longer needed
rm glad_gcta_full.part_10*

```

## Adjust for incomplete tagging of causal SNPs

-- grm adj https://yanglab.westlake.edu.cn/software/gcta/#ManipulatingtheGRM see Yang et al. 2010 Nat Genet for details  
```{bash adj gcta, eval = F}

cd ~/credentials
workdir=$(sed '2q;d' paths.config.sh) 

rm ${workdir}scripts/adj_grm.sh
cat <<'EOT'>> ${workdir}scripts/adj_grm.sh
#!/bin/bash -l
#SBATCH --partition brc,shared
#SBATCH --time=0-48:00:00
#SBATCH --nodes=1
#SBATCH --ntasks=8
#SBATCH --mem-per-cpu=19G
#SBATCH --job-name=adj_grm
#SBATCH --output=../output/adj_grm_%j.out

gendatapath=$(sed '1q;d' paths.config.sh) 
workdir=$(sed '2q;d' paths.config.sh) 
gcta=$(sed '5q;d' paths.config.sh)

${gcta} \
--grm ${workdir}results/gcta/glad_gcta_extraqc_full \
--grm-adj 0 \
--make-grm \
--out ${workdir}results/gcta/glad_gcta_extraqc_full.adj \

EOT

sbatch ${workdir}scripts/adj_grm.sh

# remove the previous, unadjusted grm
rm glad_gcta_extraqc_full.grm.*

```

## Relatedness cutoff

--grm-cutoff. Remove one of a pair of individuals with estimated relatedness larger than the specified cut-off value. GCTA selectively removes individuals to maximize the remaining sample size rather than doing it at random. To filter out individuals with IBD (pi-hat) >= 0.025 (stringent threshold recommended GCTA GREML tutorial) use --grm-cutoff 0.025. This may be too stringent e.g. https://gcta.freeforums.net/thread/260/grm-cutoff-move-most-samples ; 0.05 will suffice. 
```{bash relatedness cut-off, eval = F}

cd ~/credentials
workdir=$(sed '2q;d' paths.config.sh) 

rm ${workdir}scripts/rel_cut_grm.sh
cat <<'EOT'>> ${workdir}scripts/rel_cut_grm.sh
#!/bin/bash -l
#SBATCH --partition brc,shared
#SBATCH --time=0-48:00:00
#SBATCH --nodes=1
#SBATCH --ntasks=8
#SBATCH --mem-per-cpu=19G
#SBATCH --job-name=rel_cut_grm
#SBATCH --output=../output/rel_cut_grm_%j.out

gendatapath=$(sed '1q;d' paths.config.sh) 
workdir=$(sed '2q;d' paths.config.sh) 
gcta=$(sed '5q;d' paths.config.sh)

# remove related subjects using grm-cutoff 0.05:
${gcta} \
--grm ${workdir}results/gcta/glad_gcta_extraqc_full.adj \
--grm-cutoff 0.05 \
--make-grm \
--out ${workdir}results/gcta/glad_gcta_extraqc_full.adj.unrel05 \

EOT

sbatch ${workdir}scripts/rel_cut_grm.sh

```
After pruning the GRM, there are 16757 individuals *(373 individuals removed)*

# GCTA GREML - h2

Genome-based restricted maximum likelihood estimating genetic variance using genome-wide SNPs  

## Covariates files
Create separate continuous and categorical covariates files for GCTA GREML, plus pheno file without header  
Format: no header line; columns are family ID, individual ID and discrete covariates
```{bash awk cols for cov files, eval = F}

awk 'NR>1{print $1, $2, $13, $14}' ${workdir}input/PHQ9GAD7WSASworkimp_pheno_cov_batch_pcs_qcd_sexexcl_ibdexcl_MS.txt > ${workdir}input/sex_batch_covs.txt

awk 'NR>1{print $1, $2, $11, $12, $15, $16, $17, $18, $19, $20, $21, $22, $23, $24}' ${workdir}input/PHQ9GAD7WSASworkimp_pheno_cov_batch_pcs_qcd_sexexcl_ibdexcl_MS.txt > ${workdir}input/age_agesq_pcs_covs.txt

awk 'NR>1' ${workdir}input/PHQ9GAD7WSASworkimp_pheno_cov_batch_pcs_qcd_sexexcl_ibdexcl_MS.txt > PHQ9GAD7WSASworkimp_pheno_cov_batch_pcs_qcd_sexexcl_ibdexcl_MS_nohead.txt

```

## PHQ9
```{bash phq9 gcta greml, eval = F}

cd ~/credentials
workdir=$(sed '2q;d' paths.config.sh) 

rm ${workdir}scripts/GCTAGREML_h2_PHQ9_sumstats.sh
cat <<'EOT'>> ${workdir}scripts/GCTAGREML_h2_PHQ9_sumstats.sh
#!/bin/bash -l
#SBATCH --partition cpu
#SBATCH --time=0-48:00:00
#SBATCH --nodes=1
#SBATCH --ntasks=8
#SBATCH --mem-per-cpu=19G
#SBATCH --job-name=greml_h2
#SBATCH --output=../output/gcta_greml_phq9_extraqc_unpruned_%j.out

gendatapath=$(sed '1q;d' paths.config.sh) 
workdir=$(sed '2q;d' paths.config.sh) 
gcta=$(sed '5q;d' paths.config.sh)

${gcta} \
--reml \
--grm ${workdir}results/gcta/glad_gcta_extraqc_full.adj.unrel05 \
--pheno ${workdir}input/PHQ9GAD7WSASwork_pheno_cov_batch_pcs_qcd_sexexcl_ibdexcl_MS_nohead.txt \
--qcovar ${workdir}input/age_agesq_pcs_covs.txt \
--covar ${workdir}input/sex_batch_covs.txt \
--out ${workdir}results/gcta/phq9_greml_extraqc_full_adj_unrel05_h2 \
--thread-num 10

EOT

sbatch ${workdir}scripts/GCTAGREML_h2_PHQ9_sumstats.sh

```

## GAD7
```{bash gad7 gcta greml, eval = F}

cd ~/credentials
workdir=$(sed '2q;d' paths.config.sh) 

rm ${workdir}scripts/GCTAGREML_h2_GAD7_sumstats.sh
cat <<'EOT'>> ${workdir}scripts/GCTAGREML_h2_GAD7_sumstats.sh
#!/bin/bash -l
#SBATCH --partition brc,shared
#SBATCH --time=0-48:00:00
#SBATCH --nodes=1
#SBATCH --ntasks=8
#SBATCH --mem-per-cpu=19G
#SBATCH --job-name=gad7_greml_h2
#SBATCH --output=../output/gcta_greml_gad7_extraqc_unpruned_%j.out

gendatapath=$(sed '1q;d' paths.config.sh) 
workdir=$(sed '2q;d' paths.config.sh) 
gcta=$(sed '5q;d' paths.config.sh)

${gcta} \
--reml \
--grm ${workdir}results/gcta/glad_gcta_extraqc_full.adj.unrel05 \
--pheno ${workdir}input/PHQ9GAD7WSASwork_pheno_cov_batch_pcs_qcd_sexexcl_ibdexcl_MS_nohead.txt \
--mpheno 2 \
--qcovar ${workdir}input/age_agesq_pcs_covs.txt \
--covar ${workdir}input/sex_batch_covs.txt \
--out ${workdir}results/gcta/gad7_greml_extraqc_full_adj_unrel05_h2 \
--thread-num 10

EOT

sbatch ${workdir}scripts/GCTAGREML_h2_GAD7_sumstats.sh

```

## Imputed WSAS
Main WSAS pheno used in manuscript: 5-item score with one of the five items individual mean imputed where missing
```{bash imp wsas gcta greml, eval = F}

cd ~/credentials
workdir=$(sed '2q;d' paths.config.sh) 

rm ${workdir}scripts/GCTAGREML_h2_wsas_imp_sumstats.sh
cat <<'EOT'>> ${workdir}scripts/GCTAGREML_h2_wsas_imp_sumstats.sh
#!/bin/bash -l
#SBATCH --partition brc,shared
#SBATCH --time=0-48:00:00
#SBATCH --nodes=1
#SBATCH --ntasks=8
#SBATCH --mem-per-cpu=19G
#SBATCH --job-name=wsas_imp_greml_h2
#SBATCH --output=../output/gcta_greml_wsas_imp_extraqc_unpruned_%j.out

gendatapath=$(sed '1q;d' paths.config.sh) 
workdir=$(sed '2q;d' paths.config.sh) 
gcta=$(sed '5q;d' paths.config.sh)

${gcta} \
--reml \
--grm ${workdir}results/gcta/glad_gcta_extraqc_full.adj.unrel05 \
--pheno ${workdir}input/PHQ9GAD7WSASworkimp_pheno_cov_batch_pcs_qcd_sexexcl_ibdexcl_MS_nohead.txt \
--mpheno 7 \
--qcovar ${workdir}input/age_agesq_pcs_covs.txt \
--covar ${workdir}input/sex_batch_covs.txt \
--out ${workdir}results/gcta/wsas_imp_greml_extraqc_full_adj_unrel05_h2 \
--thread-num 10

EOT

sbatch ${workdir}scripts/GCTAGREML_h2_wsas_imp_sumstats.sh

```

## Complete case WSAS (WSAS-5-c)
5-item WSAS score
```{bash WSAS-5-c gcta greml, eval = F}

cd ~/credentials
workdir=$(sed '2q;d' paths.config.sh) 

rm ${workdir}scripts/GCTAGREML_h2_WSAS_sumstats.sh
cat <<'EOT'>> ${workdir}scripts/GCTAGREML_h2_WSAS_sumstats.sh
#!/bin/bash -l
#SBATCH --partition brc,shared
#SBATCH --time=0-48:00:00
#SBATCH --nodes=1
#SBATCH --ntasks=8
#SBATCH --mem-per-cpu=19G
#SBATCH --job-name=wsas_greml_h2
#SBATCH --output=../output/gcta_greml_wsas_extraqc_unpruned_%j.out

gendatapath=$(sed '1q;d' paths.config.sh) 
workdir=$(sed '2q;d' paths.config.sh) 
gcta=$(sed '5q;d' paths.config.sh)

${gcta} \
--reml \
--grm ${workdir}results/gcta/glad_gcta_extraqc_full.adj.unrel05 \
--pheno ${workdir}input/PHQ9GAD7WSASwork_pheno_cov_batch_pcs_qcd_sexexcl_ibdexcl_MS_nohead.txt \
--mpheno 3 \
--qcovar ${workdir}input/age_agesq_pcs_covs.txt \
--covar ${workdir}input/sex_batch_covs.txt \
--out ${workdir}results/gcta/wsas_greml_extraqc_full_adj_unrel05_h2 \
--thread-num 10

EOT

sbatch ${workdir}scripts/GCTAGREML_h2_WSAS_sumstats.sh

```
  
## WSAS-4
4-item WSAS score
```{bash WSAS-4 gcta greml, eval = F}

cd ~/credentials
workdir=$(sed '2q;d' paths.config.sh) 

rm ${workdir}scripts/GCTAGREML_h2_rWSAS4_sumstats.sh
cat <<'EOT'>> ${workdir}scripts/GCTAGREML_h2_rWSAS4_sumstats.sh
#!/bin/bash -l
#SBATCH --partition brc,shared
#SBATCH --time=0-48:00:00
#SBATCH --nodes=1
#SBATCH --ntasks=8
#SBATCH --mem-per-cpu=19G
#SBATCH --job-name=rwsas4_greml_h2
#SBATCH --output=../output/gcta_greml_rwsas4_extraqc_unpruned_%j.out

gendatapath=$(sed '1q;d' paths.config.sh) 
workdir=$(sed '2q;d' paths.config.sh) 
gcta=$(sed '5q;d' paths.config.sh)

${gcta} \
--reml \
--grm ${workdir}results/gcta/glad_gcta_extraqc_full.adj.unrel05 \
--pheno ${workdir}input/PHQ9GAD7WSASwork_pheno_cov_batch_pcs_qcd_sexexcl_ibdexcl_MS_nohead.txt \
--mpheno 4 \
--qcovar ${workdir}input/age_agesq_pcs_covs.txt \
--covar ${workdir}input/sex_batch_covs.txt \
--out ${workdir}results/gcta/rwsas4_greml_extraqc_full_adj_unrel05_h2 \
--thread-num 10

EOT

sbatch ${workdir}scripts/GCTAGREML_h2_rWSAS4_sumstats.sh

```

## Completers' WSAS-4 
4-item WSAS score only for Ps with all 5 items
```{bash compl WSAS-4 gcta greml, eval = F}

cd ~/credentials
workdir=$(sed '2q;d' paths.config.sh) 

rm ${workdir}scripts/GCTAGREML_h2_complrWSAS4_sumstats.sh
cat <<'EOT'>> ${workdir}scripts/GCTAGREML_h2_complrWSAS4_sumstats.sh
#!/bin/bash -l
#SBATCH --partition brc,shared
#SBATCH --time=0-48:00:00
#SBATCH --nodes=1
#SBATCH --ntasks=8
#SBATCH --mem-per-cpu=19G
#SBATCH --job-name=complrwsas4_greml_h2
#SBATCH --output=../output/gcta_greml_complrwsas4_extraqc_unpruned_%j.out

gendatapath=$(sed '1q;d' paths.config.sh) 
workdir=$(sed '2q;d' paths.config.sh) 
gcta=$(sed '5q;d' paths.config.sh)

${gcta} \
--reml \
--grm ${workdir}results/gcta/glad_gcta_extraqc_full.adj.unrel05 \
--pheno ${workdir}input/PHQ9GAD7WSASwork_pheno_cov_batch_pcs_qcd_sexexcl_ibdexcl_MS_nohead.txt \
--mpheno 5 \
--qcovar ${workdir}input/age_agesq_pcs_covs.txt \
--covar ${workdir}input/sex_batch_covs.txt \
--out ${workdir}results/gcta/complrwsas4_greml_extraqc_full_adj_unrel05_h2 \
--thread-num 10

EOT

sbatch ${workdir}scripts/GCTAGREML_h2_complrWSAS4_sumstats.sh

```

## WSAS work item
```{bash wsas work gcta greml, eval = F}

cd ~/credentials
workdir=$(sed '2q;d' paths.config.sh) 

rm ${workdir}scripts/GCTAGREML_h2_work_sumstats.sh
cat <<'EOT'>> ${workdir}scripts/GCTAGREML_h2_work_sumstats.sh
#!/bin/bash -l
#SBATCH --partition brc,shared
#SBATCH --time=0-48:00:00
#SBATCH --nodes=1
#SBATCH --ntasks=8
#SBATCH --mem-per-cpu=19G
#SBATCH --job-name=work_greml_h2
#SBATCH --output=../output/gcta_greml_work_extraqc_unpruned_%j.out

gendatapath=$(sed '1q;d' paths.config.sh) 
workdir=$(sed '2q;d' paths.config.sh) 
gcta=$(sed '5q;d' paths.config.sh)

${gcta} \
--reml \
--grm ${workdir}results/gcta/glad_gcta_extraqc_full.adj.unrel05 \
--pheno ${workdir}input/PHQ9GAD7WSASwork_pheno_cov_batch_pcs_qcd_sexexcl_ibdexcl_MS_nohead.txt \
--mpheno 6 \
--qcovar ${workdir}input/age_agesq_pcs_covs.txt \
--covar ${workdir}input/sex_batch_covs.txt \
--out ${workdir}results/gcta/work_greml_extraqc_full_adj_unrel05_h2 \
--thread-num 10

EOT

sbatch ${workdir}scripts/GCTAGREML_h2_work_sumstats.sh

```

## Table
```{r Table gcta greml h2}
                        
trait = c("PHQ9", "GAD7", "WSAS")
obs.h2 = c(0.191679, 0.172675, 0.110836)
obs.h2.se = c(0.035064, 0.034880, 0.033342)
p.val = c(0.0000000055016, 0.00000016956, 0.00018258)
n.size = c(16709, 16725, 16736)
lower.ci = c(obs.h2-(1.959964*obs.h2.se))
upper.ci = c(obs.h2+(1.959964*obs.h2.se))
greml.h2 <- data.frame(trait, obs.h2, obs.h2.se, lower.ci, upper.ci, p.val, n.size)
greml.h2.table <- greml.h2%>%
  mutate(across(where(is.numeric), formatv, digit = 1, nsmall = 3))
greml.h2.table%>%kable(caption = "GCTA-GREML h2 Output",
        col.names = c("Trait", "Observed h2", "h2 SE", "Lower 95% CI", "Upper 95% CI", "p", "n"),
        escape = F,
        digits = c(0, rep(3, 4), 4, 0),
        align = "lcccccc")%>%
  kable_styling()

```

## Figure
```{r Fig greml h2 estimates}

greml.h2 <- greml.h2%>%
  mutate(trait = fct_relevel(trait, c("PHQ9", "GAD7", "WSAS")))

ggplot(data=greml.h2, 
       aes(x=trait,
           y=obs.h2,
           colour=trait)) +
  geom_point(position=position_dodge(width=.5), 
             size=3) +
  geom_errorbar(aes(ymin=obs.h2-(1.959964*obs.h2.se), ymax=obs.h2 + (1.959964*obs.h2.se)),
                position=position_dodge(0.5),
                width=.2, 
                size=0.6) +
  scale_fill_manual(values = my.palette) +
  scale_color_manual(values = my.palette) +
  scale_y_continuous(breaks = seq(0, 0.30, 0.05), limits = c(-0.01, 0.305)) +
  labs(y = "GCTA-GREML h2") +
  my_theme +
  theme(legend.position = "none",
        panel.grid.major.y = element_line(linetype = "dotted",
                                          colour = "gray"))

```

## Table & plot with complete case WSAS-5-c and 4-item WSAS
```{r Table gcta greml h2 WSAS-5-c WSAS-4}
                        
trait = c("PHQ9", "GAD7", "WSAS-5-c", "WSAS-4")
obs.h2 = c(0.191679, 0.172675, 0.133479, 0.110671)
obs.h2.se = c(0.035064, 0.034880, 0.038021, 0.033288)
p.val = c(0.0000000055016, 0.00000016956, 0.000096283, 0.0001591)
n.size = c(16709, 16725, 14744, 16710)
lower.ci = c(obs.h2-(1.959964*obs.h2.se))
upper.ci = c(obs.h2+(1.959964*obs.h2.se))
greml.h2 <- data.frame(trait, obs.h2, obs.h2.se, lower.ci, upper.ci, p.val, n.size)
greml.h2.table <- greml.h2%>%
  mutate(across(where(is.numeric), formatv, digit = 1, nsmall = 3))
greml.h2.table%>%kable(caption = "GCTA-GREML h2 Output",
        col.names = c("Trait", "Observed h2", "h2 SE", "Lower 95% CI", "Upper 95% CI", "p", "n"),
        escape = F,
        digits = c(0, rep(3, 4), 4, 0),
        align = "lccccc")%>%
  kable_styling()

```

```{r Fig greml h2 estimates WSAS-5-c WSAS-4}

greml.h2 <- greml.h2%>%
  mutate(trait = fct_relevel(trait, c("PHQ9", "GAD7", "WSAS-5-c", "WSAS-4")))

ggplot(data=greml.h2, 
       aes(x=trait,
           y=obs.h2,
           colour=trait)) +
  geom_point(position=position_dodge(width=.5), 
             size=3) +
  geom_errorbar(aes(ymin=obs.h2-(1.959964*obs.h2.se), ymax=obs.h2 + (1.959964*obs.h2.se)),
                position=position_dodge(0.5),
                width=.2,
                size=0.6) +
  scale_fill_manual(values = my.palette) +
  scale_color_manual(values = my.palette) +
  scale_y_continuous(breaks = seq(0, 0.30, 0.05), limits = c(-0.01, 0.305)) +
  labs(y = "GCTA-GREML h2") +
  my_theme +
  theme(legend.position = "none",
        panel.grid.major.y = element_line(linetype = "dotted",
                                          colour = "gray"))

```

# Bivariate GREML - rg

Be careful with reml-bivar numbers indicating pheno columns as these change - have added sensitivity variables to pheno file  
phq9 = phq9.sum_score, #pheno 1  
gad7 = gad7.sum_score, #pheno 2  
wsas = wsas.sum_score, #pheno 3; wsas sumscores (will be NA if missing any item)  
rwsas4 = wsas.4item_sum_score, #pheno 4; 4-item wsas sumscore without work item  
complwsas_rwsas4, #pheno 5; 4-item wsas sumscore only for Ps with 5 items complete  
incomplwsas_rwsas4, #pheno 6; 4-item wsas sumscore only for Ps missing an item  
wsas_work = wsas.work_study_na_numeric, #pheno 7; only the score for the work item  
wsas_imp = wsas.sum_score_imp, #pheno 8; wsas sumscores observed or imputed total for Ps missing one item

--reml-bivar-lrt-rg 0 provides p-value of difference from rg = 0, without it you get the same rg but no p-value

## PHQ9 GAD7
```{bash phq9 gad7 bivariate greml, eval = F}

cd ~/credentials
workdir=$(sed '2q;d' paths.config.sh) 

rm ${workdir}scripts/BIGREML_rg_PHQ9GAD7.sh
cat <<'EOT'>> ${workdir}scripts/BIGREML_rg_PHQ9GAD7.sh
#!/bin/bash -l
#SBATCH --partition cpu
#SBATCH --time=0-48:00:00
#SBATCH --nodes=1
#SBATCH --ntasks=8
#SBATCH --mem-per-cpu=19G
#SBATCH --job-name=greml_rg
#SBATCH --output=../output/greml_rg_phq9_gad7_covs_%j.out

workdir=$(sed '2q;d' paths.config.sh) 
gcta=$(sed '5q;d' paths.config.sh)

${gcta} \
--reml-bivar \
--grm ${workdir}results/gcta/glad_gcta_extraqc_full.adj.unrel05 \
--pheno ${workdir}input/PHQ9GAD7WSASworkimp_pheno_cov_batch_pcs_qcd_sexexcl_ibdexcl_MS_nohead.txt \
--qcovar ${workdir}input/age_agesq_pcs_covs.txt \
--covar ${workdir}input/sex_batch_covs.txt \
--reml-bivar-lrt-rg 0 \
--out ${workdir}results/gcta/phq9_gad7_greml_extraqc_full_adj_unrel05_covs_LRT0_rg \
--thread-num 10

EOT

sbatch ${workdir}scripts/BIGREML_rg_PHQ9GAD7.sh

```

## PHQ9 Imputed-WSAS
Comparison with WSAS phenotype used in manuscript
```{bash phq9 wsas_imp bivariate greml, eval = F}

cd ~/credentials
workdir=$(sed '2q;d' paths.config.sh) 

rm ${workdir}scripts/BIGREML_rg_PHQ9WSAS_imp_sumstats.sh
cat <<'EOT'>> ${workdir}scripts/BIGREML_rg_PHQ9WSAS_imp_sumstats.sh
#!/bin/bash -l
#SBATCH --partition cpu
#SBATCH --time=0-48:00:00
#SBATCH --nodes=1
#SBATCH --ntasks=8
#SBATCH --mem-per-cpu=19G
#SBATCH --job-name=greml_rg
#SBATCH --output=../output/greml_rg_phq9_wsasimp_covs_%j.out

workdir=$(sed '2q;d' paths.config.sh) 
gcta=$(sed '5q;d' paths.config.sh)

${gcta} \
--reml-bivar 1 8 \
--grm ${workdir}results/gcta/glad_gcta_extraqc_full.adj.unrel05 \
--pheno ${workdir}input/PHQ9GAD7WSASworkimp_pheno_cov_batch_pcs_qcd_sexexcl_ibdexcl_MS_nohead.txt \
--qcovar ${workdir}input/age_agesq_pcs_covs.txt \
--covar ${workdir}input/sex_batch_covs.txt \
--reml-bivar-lrt-rg 0 \
--out ${workdir}results/gcta/phq9_wsas_imp_greml_extraqc_full_adj_unrel05_covs_LRT0_rg \
--thread-num 10

EOT

sbatch ${workdir}scripts/BIGREML_rg_PHQ9WSAS_imp_sumstats.sh

```


### p-value for test of rg significantly different from 1 
Change reml-bivar-lrt-rg to 1
```{bash PHQ9 WSAS imp diff from 1}

cd ~/credentials
workdir=$(sed '2q;d' paths.config.sh) 

rm ${workdir}scripts/BIGREML_rg_PHQ9WSAS_imp_sumstats.sh
cat <<'EOT'>> ${workdir}scripts/BIGREML_rg_PHQ9WSAS_imp_sumstats.sh
#!/bin/bash -l
#SBATCH --partition cpu
#SBATCH --time=0-48:00:00
#SBATCH --nodes=1
#SBATCH --ntasks=8
#SBATCH --mem-per-cpu=19G
#SBATCH --job-name=greml_rg
#SBATCH --output=../output/greml_rg_phq9_wsasimp_covs_%j.out

workdir=$(sed '2q;d' paths.config.sh) 
gcta=$(sed '5q;d' paths.config.sh)

${gcta} \
--reml-bivar 1 8 \
--grm ${workdir}results/gcta/glad_gcta_extraqc_full.adj.unrel05 \
--pheno ${workdir}input/PHQ9GAD7WSASworkimp_pheno_cov_batch_pcs_qcd_sexexcl_ibdexcl_MS_nohead.txt \
--qcovar ${workdir}input/age_agesq_pcs_covs.txt \
--covar ${workdir}input/sex_batch_covs.txt \
--reml-bivar-lrt-rg 1 \
--out ${workdir}results/gcta/phq9_wsas_imp_greml_extraqc_full_adj_unrel05_covs_LRT1_rg \
--thread-num 10

EOT

sbatch ${workdir}scripts/BIGREML_rg_PHQ9WSAS_imp_sumstats.sh

```


### Estimate bivariate SNP heritability; proportion of phenotypic correlation explained by rg  

Bivariate SNP Hh2 from Morris et al., 2020 = (rg x sqrt(h2A) x sqrt(h2B))/rp  
Used SNP h2 ests from bivariate GREML rather than original h2 ests - differ slightly due to variants present between traits 

Standard error estimated via simulations as per Morris et al., 2020:  
"estimates were derived from entering the GCTA estimated components into Eq. 4 above. SEs for the bivariate heritability estimates were estimated using simulations. First, we simulated 1 million  observations for each input parameter (the two heritability terms,  genetic correlation and phenotypic correlation) as normally distributed with a mean value corresponding to the point estimate and SD corresponding to the SE obtained from GCTA. Bivariate heritability was estimated for each observation and the SD of all 1 million estimates taken as the estimated error for the point estimate. This  approach to calculating SEs assumes no covariance between the input parameters. Nonzero covariances would produce smaller SEs, and therefore, this approach can be considered to provide conservative estimates". Used phenotypic correlation from descriptives however this will slightly differ due to Ps included in the GCTA analysis  

```{r biv snp h2 phq9 wsas_imp}

phqwsas_bivh2 <- (0.873797*(sqrt(0.191080))*(sqrt(0.111659)))/0.6321958

#simulate standard error
h_phq_sim <- rtruncnorm(n = 1000000, a=0, b=1, mean = 0.191080, sd = 0.035060)
h_wsas_imp_sim <- rtruncnorm(n = 1000000, a=0, b=1, mean = 0.111659, sd = 0.033076)
rg_phqwsas_sim <- rtruncnorm(n = 1000000, a=0, b=1, mean = 0.873797, sd = 0.087387)
rp_phqwsas_sim <- rtruncnorm(n = 1000000, a=0, b=1, mean = 0.6321958, sd = 0.00593) #SD is SE here as well; see Suppl Table 6 Morris et al., 2020  

sim_phqwsas_bivh2<-(rg_phqwsas_sim*(sqrt(h_phq_sim))*(sqrt(h_wsas_imp_sim)))/rp_phqwsas_sim

```
Bivariate SNP h2 is estimated as `r round(phqwsas_bivh2, 3)`, simulated bivariate SNP h2 is `r round(mean(sim_phqwsas_bivh2), 3)` and simulated standard error is `r round(sd(sim_phqwsas_bivh2), 3)`. The confidence interval using the simulated values is `r mean(sim_phqwsas_bivh2)-((sd(sim_phqwsas_bivh2))*1.959964)` to `r mean(sim_phqwsas_bivh2)+((sd(sim_phqwsas_bivh2))*1.959964)`

## GAD7 Imputed-WSAS
```{bash gad7 wsas_imp bivariate greml, eval = F}

cd ~/credentials
workdir=$(sed '2q;d' paths.config.sh) 

rm ${workdir}scripts/BIGREML_rg_GAD7WSAS_imp_sumstats.sh
cat <<'EOT'>> ${workdir}scripts/BIGREML_rg_GAD7WSAS_imp_sumstats.sh
#!/bin/bash -l
#SBATCH --partition cpu
#SBATCH --time=0-48:00:00
#SBATCH --nodes=1
#SBATCH --ntasks=8
#SBATCH --mem-per-cpu=19G
#SBATCH --job-name=greml_rg
#SBATCH --output=../output/greml_rg_gad7_wsasimp_covs_%j.out

workdir=$(sed '2q;d' paths.config.sh) 
gcta=$(sed '5q;d' paths.config.sh)

${gcta} \
--reml-bivar 2 8 \
--grm ${workdir}results/gcta/glad_gcta_extraqc_full.adj.unrel05 \
--pheno ${workdir}input/PHQ9GAD7WSASworkimp_pheno_cov_batch_pcs_qcd_sexexcl_ibdexcl_MS_nohead.txt \
--qcovar ${workdir}input/age_agesq_pcs_covs.txt \
--covar ${workdir}input/sex_batch_covs.txt \
--reml-bivar-lrt-rg 0 \
--out ${workdir}results/gcta/gad7_wsas_imp_greml_extraqc_full_adj_unrel05_covs_LRT0_rg \
--thread-num 10

EOT

sbatch ${workdir}scripts/BIGREML_rg_GAD7WSAS_imp_sumstats.sh

```

### p-value for test of rg significantly different from 1

```{bash GAD7 WSAS imp diff from 1}

cd ~/credentials
workdir=$(sed '2q;d' paths.config.sh) 

rm ${workdir}scripts/BIGREML_rg_GAD7WSAS_imp_sumstats.sh
cat <<'EOT'>> ${workdir}scripts/BIGREML_rg_GAD7WSAS_imp_sumstats.sh
#!/bin/bash -l
#SBATCH --partition cpu
#SBATCH --time=0-48:00:00
#SBATCH --nodes=1
#SBATCH --ntasks=8
#SBATCH --mem-per-cpu=19G
#SBATCH --job-name=greml_rg
#SBATCH --output=../output/greml_rg_gad7_wsasimp_covs_%j.out

workdir=$(sed '2q;d' paths.config.sh) 
gcta=$(sed '5q;d' paths.config.sh)

${gcta} \
--reml-bivar 2 8 \
--grm ${workdir}results/gcta/glad_gcta_extraqc_full.adj.unrel05 \
--pheno ${workdir}input/PHQ9GAD7WSASworkimp_pheno_cov_batch_pcs_qcd_sexexcl_ibdexcl_MS_nohead.txt \
--qcovar ${workdir}input/age_agesq_pcs_covs.txt \
--covar ${workdir}input/sex_batch_covs.txt \
--reml-bivar-lrt-rg 1 \
--out ${workdir}results/gcta/gad7_wsas_imp_greml_extraqc_full_adj_unrel05_covs_LRT1_rg \
--thread-num 10

EOT

sbatch ${workdir}scripts/BIGREML_rg_GAD7WSAS_imp_sumstats.sh

```

### Estimate bivariate SNP heritability; proportion of phenotypic correlation explained by rg  

```{r biv snp h2 gad7 wsas_imp}

gadwsas_bivh2 <- (0.794456*(sqrt(0.171122))*(sqrt(0.113121)))/0.4949930

#simulate standard error
h_gad_sim <- rtruncnorm(n = 1000000, a=0, b=1, mean = 0.171122, sd = 0.034875)
h_wsas_imp_sim <- rtruncnorm(n = 1000000, a=0, b=1, mean = 0.113121, sd = 0.033074)
rg_gadwsas_sim <- rtruncnorm(n = 1000000, a=0, b=1, mean = 0.794456, sd = 0.116518)
rp_gadwsas_sim <- rtruncnorm(n = 1000000, a=0, b=1, mean = 0.4949930, sd = 0.006649648) #SD is SE here as well; see Suppl Table 6 Morris et al.  

sim_gadwsas_bivh2<-(rg_gadwsas_sim*(sqrt(h_gad_sim))*(sqrt(h_wsas_imp_sim)))/rp_gadwsas_sim

```
Bivariate SNP h2 is estimated as `r round(gadwsas_bivh2, 3)`, simulated bivariate SNP h2 is `r round(mean(sim_gadwsas_bivh2), 3)` and simulated standard error is `r round(sd(sim_gadwsas_bivh2), 3)`. The confidence interval using the simulated values is `r mean(sim_gadwsas_bivh2)-((sd(sim_gadwsas_bivh2))*1.959964)` to `r mean(sim_gadwsas_bivh2)+((sd(sim_gadwsas_bivh2))*1.959964)`

## Tables

```{r gcta greml rg table}

#rg
PHQ9.rg = c(1, 0.864929, 0.873797)                                    
GAD7.rg = c(0.864929, 1, 0.794456)        
WSAS.rg = c(0.873797, 0.794456, 1)
greml.rg = rbind(PHQ9.rg, GAD7.rg, WSAS.rg)
colnames(greml.rg) = c("PHQ9", "GAD7", "WSAS")
rownames(greml.rg) = c("PHQ9", "GAD7", "WSAS")
greml.rg%>%kable(caption = "rgs",
                   digits = c(rep(2, 3)))%>%
  kable_styling()

#SE
PHQ9.rg.se = c(0, 0.057380, 0.087387)
GAD7.rg.se = c(0.057380, 0, 0.116518)
WSAS.rg.se = c(0.087387, 0.116518, 0)
greml.rg.se = rbind(PHQ9.rg.se, GAD7.rg.se, WSAS.rg.se)
colnames(greml.rg.se) = c("PHQ9", "GAD7", "WSAS")
greml.rg.se%>%kable(caption = "rg SEs",
                   digits = c(rep(2, 3)))%>%
  kable_styling()

#95% CI
PHQ9.rg.lci = round(PHQ9.rg - (1.959964*PHQ9.rg.se),2)
PHQ9.rg.uci = round(PHQ9.rg + (1.959964*PHQ9.rg.se),2)
GAD7.rg.lci = round(GAD7.rg - (1.959964*GAD7.rg.se),2)
GAD7.rg.uci = round(GAD7.rg + (1.959964*GAD7.rg.se),2)
WSAS.rg.lci = round(WSAS.rg - (1.959964*WSAS.rg.se),2)
WSAS.rg.uci = round(WSAS.rg + (1.959964*WSAS.rg.se),2)
PHQ9.rg.ci = paste0(PHQ9.rg.lci, " - ", PHQ9.rg.uci)
GAD7.rg.ci = paste0(GAD7.rg.lci, " - ", GAD7.rg.uci)
WSAS.rg.ci = paste0(WSAS.rg.lci, " - ", WSAS.rg.uci)
greml.rg.ci = cbind(PHQ9.rg.ci, GAD7.rg.ci, WSAS.rg.ci)
rownames(greml.rg.ci) = c("PHQ9", "GAD7", "WSAS")
colnames(greml.rg.ci) = c("PHQ9", "GAD7", "WSAS")
greml.rg.ci%>%
  kable(caption = "rg CIs")%>%
  kable_styling()

```

## Figure

```{r gcta greml rg fig}

greml.rg = greml.rg%>%
  as.data.frame(.)%>%
  rownames_to_column(var = "measure1")%>%
    pivot_longer(cols = c(PHQ9, GAD7, WSAS), 
               names_to = "measure2", 
               values_to = "value")%>%
  mutate(across(where(is.numeric), round, 3))%>%
  filter(!is.na(value))

greml.rg = greml.rg %>%
  mutate(measure1 = factor(measure1, levels = c("PHQ9", "GAD7", "WSAS")),
         measure2 = factor(measure2, levels = c("PHQ9", "GAD7", "WSAS")))

greml.rg%>%
  arrange(measure1, measure2) %>%
  group_by(measure1) %>%
  filter(row_number() >= which(measure1 == measure2)) %>%
ggplot(., 
       aes(measure2, measure1, fill = value))+
  geom_tile(color = "white")+
  scale_fill_gradient2(low = "blue",
                       high = "red",
                       mid = "white",
                       midpoint = 0,
                       limit = c(-1,1),
                       space = "Lab",
                       name="Pearson\nCorrelation") +
  coord_fixed() + 
geom_text(aes(measure2, measure1, label = value), 
          color = "black",
          size = 5) +
  labs(title = "Bivariate-GREML estimated rg") +
theme(axis.title = element_blank(),
      axis.text = element_text(colour="black",
                               size = 12,
                               angle = 30),
  panel.grid.major = element_blank(),
  panel.border = element_blank(),
  panel.background = element_blank(),
  axis.ticks = element_blank(),
  legend.justification = c(1, 0),
  legend.position = c(0.45, 0.7),
  legend.direction = "horizontal")+
  guides(fill = guide_colorbar(barwidth = 7,
                               barheight = 1,
                title.position = "top",
                title.hjust = 0.5))

```

Alternative plot: point with error bars *Used in paper*

duplicates information e.g. PHQ9 with GAD7 and GAD7 with PHQ9 but easier to read maybe?
```{r data prep gcta rg point error bar}

measure1 <- factor(rep(c("PHQ9", "GAD7", "WSAS"), each = 3, times = 2))
measure2 <- factor(rep(c("PHQ9", "GAD7", "WSAS"), 6))
corr.type <- factor(rep(c("genetic", "phenotypic"), each = 9))

rg <- c(1, 0.864929, 0.873797,
        0.864929, 1, 0.794456,
        0.873797, 0.794456, 1)

rg.se <- c(0, 0.057380, 0.087387, 
           0.057380, 0, 0.116518,
           0.087387, 0.116518, 0)

rg.lci <- rg-(1.959964*rg.se)
rg.uci <- rg+(1.959964*rg.se)
#rg.p
rgp <- c(rg, 
         1, 0.693, 0.632, 
         0.693, 1, 0.495, 
         0.632, 0.495, 1)
rgp.lci <- c(rg.lci, 
             1, 0.685, 0.623,
             0.685, 1, 0.484, 
             0.623, 0.484, 1)
rgp.uci <- c(rg.uci,
             1, 0.701, 0.641, 
             0.701, 1, 0.506, 
             0.641, 0.506, 1)
#rgp.p <- c(rg.p, rep(0, length(rp)))
#rgp.sig <- rgp.p<(0.05/3)
rgp.plot.point <- data.frame(measure1, measure2, corr.type, rgp, rgp.lci, rgp.uci)
rgp.plot.point <- rgp.plot.point%>%
  mutate(measure1 = fct_relevel(measure1, "PHQ9", "GAD7", "WSAS"),
         measure2 = fct_relevel(measure2, "PHQ9", "GAD7", "WSAS"))%>%
   unite("corrmeasure", measure2, corr.type, sep = " ", remove = FALSE)%>%
  mutate(corrmeasure = fct_relevel(corrmeasure,
                                   c("PHQ9 genetic",
                                     "PHQ9 phenotypic",
                                     "GAD7 genetic",
                                     "GAD7 phenotypic",
                                     "WSAS genetic",
                                     "WSAS phenotypic")))

rgp.plot.point.noself = rgp.plot.point[c(2,3,4,6,7,8,11,12,13,15,16,17),]
rgp.plot.point.lower = rgp.plot.point.noself[c(1,2,4,7,8,10),]

```

```{r Plot gcta rg point error bar}

Dark2rep = c("#E7298A", "#E7298A", "#009974", "#009974", "#D95F02", "#D95F02", "#7570B3", "#7570B3")

ggplot(rgp.plot.point.noself,
       aes(x=measure1,
           y=rgp,
           group=corrmeasure, 
           colour=corrmeasure,
           shape=corrmeasure)) + 
 geom_point(position=position_dodge(width=.5),
            size=3) +
  geom_errorbar(aes(ymin=rgp.lci, ymax=rgp.uci),
       width=.1,
       position=position_dodge(0.5)) +
  scale_fill_manual(values = Dark2rep) +
  scale_color_manual(values = Dark2rep) +
  scale_shape_manual(values = c(15,0,16,1,17,2)) + #hollow/filled square = 0/15, hollow/filled circle = 1/16, hollow/filled triangle = 2/17
  scale_y_continuous(limits=c(0,1.05),
                     breaks=c(0,0.25,0.5,0.75,1)) +
  geom_hline(yintercept=as.numeric(c(0)),
             linetype="dotted",
             colour="darkgray")+
  labs(y="Correlation") + 
   guides(colour = guide_legend(nrow=2,byrow=TRUE)) + 
  my_theme +
  theme(legend.title = element_blank(),
        legend.position = "bottom",
        axis.title.x=element_blank(),
        panel.grid.major.y = element_line(linetype = "dotted",
                                          colour = "gray"))
ggsave("gcta.genpheno.corr.png", width = 18, height = 10, units = "cm", dpi=800)

```

#### For legend

```{r rg plot legend}

dark2diff <- c("#E7298A", "#009974", "#D95F02", "#E7298A", "#009974", "#D95F02")
  
rgp.plot.point <- data.frame(measure1, measure2, corr.type, rgp, rgp.lci, rgp.uci)
rgp.plot.point <- rgp.plot.point%>%
  mutate(measure1 = fct_relevel(measure1, "PHQ9", "GAD7", "WSAS"),
         measure2 = fct_relevel(measure2, "PHQ9", "GAD7", "WSAS"))%>%
   unite("corrmeasure", measure2, corr.type, sep = " ", remove = FALSE)%>%
  mutate(corrmeasure = fct_relevel(corrmeasure,
                                   c("PHQ9 genetic",
                                     "GAD7 genetic",
                                     "WSAS genetic",
                                     "PHQ9 phenotypic",
                                     "GAD7 phenotypic",
                                     "WSAS phenotypic")))

rgp.plot.point.noself = rgp.plot.point[c(2,3,4,6,7,8,11,12,13,15,16,17),]

ggplot(rgp.plot.point.noself,
       aes(x=measure1,
           y=rgp,
           group=corrmeasure, 
           colour=corrmeasure,
           shape=corrmeasure)) + 
 geom_point(position=position_dodge(width=.5),
            size=3) +
  geom_errorbar(aes(ymin=rgp.lci, ymax=rgp.uci),
       width=.1,
       position=position_dodge(0.5)) +
  scale_fill_manual(values = dark2diff) +
  scale_color_manual(values = dark2diff) +
  scale_shape_manual(values = c(15,16,17,0,1,2)) + #hollow/filled square = 0/15, hollow/filled circle = 1/16, hollow/filled triangle = 2/17
  scale_y_continuous(limits=c(0,1.02),
                     breaks=c(0,0.25,0.5,0.75,1)) +
  geom_hline(yintercept=as.numeric(c(0)),
             linetype="dotted",
             colour="darkgray")+
  labs(y="Correlation") + 
   guides(colour = guide_legend(nrow=2,byrow=TRUE)) + 
  my_theme +
  theme(legend.title = element_blank(),
        legend.position = "bottom",
        axis.title.x=element_blank(),
        panel.grid.major.y = element_line(linetype = "dotted",
                                          colour = "gray"))
ggsave("gcta.genpheno.corr.legend.png", width = 18, height = 10, units = "cm", dpi=800)

```

# Sensitivity/secondary analyses with WSAS-5-c, WSAS-4 and WSAS work item

## PHQ9 WSAS-5-c
Sensitivity rg - compare to imputed WSAS and WSAS-4 results
```{bash phq9 WSAS-5-c bivariate greml, eval = F}

cd ~/credentials
workdir=$(sed '2q;d' paths.config.sh) 

rm ${workdir}scripts/BIGREML_rg_PHQ9WSAS_sumstats.sh
cat <<'EOT'>> ${workdir}scripts/BIGREML_rg_PHQ9WSAS_sumstats.sh
#!/bin/bash -l
#SBATCH --partition cpu
#SBATCH --time=0-48:00:00
#SBATCH --nodes=1
#SBATCH --ntasks=8
#SBATCH --mem-per-cpu=19G
#SBATCH --job-name=greml_rg
#SBATCH --output=../output/greml_rg_phq9_wsas_covs_%j.out

workdir=$(sed '2q;d' paths.config.sh) 
gcta=$(sed '5q;d' paths.config.sh)

${gcta} \
--reml-bivar 1 3 \
--grm ${workdir}results/gcta/glad_gcta_extraqc_full.adj.unrel05 \
--pheno ${workdir}input/PHQ9GAD7WSASworkimp_pheno_cov_batch_pcs_qcd_sexexcl_ibdexcl_MS_nohead.txt \
--qcovar ${workdir}input/age_agesq_pcs_covs.txt \
--covar ${workdir}input/sex_batch_covs.txt \
--reml-bivar-lrt-rg 0 \
--out ${workdir}results/gcta/phq9_wsas_greml_extraqc_full_adj_unrel05_covs_LRT0_rg \
--thread-num 10

EOT

sbatch ${workdir}scripts/BIGREML_rg_PHQ9WSAS_sumstats.sh

```

## GAD7 WSAS-5-c
Sensitivity - compare to imputed WSAS and WSAS-4 results
```{bash gad7 WSAS-5-c bivariate greml, eval = F}

cd ~/credentials
workdir=$(sed '2q;d' paths.config.sh) 

rm ${workdir}scripts/BIGREML_rg_GAD7WSAS_sumstats.sh
cat <<'EOT'>> ${workdir}scripts/BIGREML_rg_GAD7WSAS_sumstats.sh
#!/bin/bash -l
#SBATCH --partition cpu
#SBATCH --time=0-48:00:00
#SBATCH --nodes=1
#SBATCH --ntasks=8
#SBATCH --mem-per-cpu=19G
#SBATCH --job-name=greml_rg
#SBATCH --output=../output/greml_rg_gad7_wsas_covs_%j.out

workdir=$(sed '2q;d' paths.config.sh) 
gcta=$(sed '5q;d' paths.config.sh)

${gcta} \
--reml-bivar 2 3 \
--grm ${workdir}results/gcta/glad_gcta_extraqc_full.adj.unrel05 \
--pheno ${workdir}input/PHQ9GAD7WSASworkimp_pheno_cov_batch_pcs_qcd_sexexcl_ibdexcl_MS_nohead.txt \
--qcovar ${workdir}input/age_agesq_pcs_covs.txt \
--covar ${workdir}input/sex_batch_covs.txt \
--reml-bivar-lrt-rg 0 \
--out ${workdir}results/gcta/gad7_wsas_greml_extraqc_full_adj_unrel05_covs_LRT0_rg \
--thread-num 10

EOT

sbatch ${workdir}scripts/BIGREML_rg_GAD7WSAS_sumstats.sh

```

## PHQ9 WSAS-4
Sensitivity - compare to imputed WSAS and WSAS-5-c results
```{bash phq9 WSAS-4 bivariate greml, eval = F}

cd ~/credentials
workdir=$(sed '2q;d' paths.config.sh) 

rm ${workdir}scripts/BIGREML_rg_PHQ9WSAS4_sumstats.sh
cat <<'EOT'>> ${workdir}scripts/BIGREML_rg_PHQ9WSAS4_sumstats.sh
#!/bin/bash -l
#SBATCH --partition cpu
#SBATCH --time=0-48:00:00
#SBATCH --nodes=1
#SBATCH --ntasks=8
#SBATCH --mem-per-cpu=19G
#SBATCH --job-name=greml_rg
#SBATCH --output=../output/greml_rg_phq9_rwsas4_covs_%j.out

workdir=$(sed '2q;d' paths.config.sh) 
gcta=$(sed '5q;d' paths.config.sh)

${gcta} \
--reml-bivar 1 4 \
--grm ${workdir}results/gcta/glad_gcta_extraqc_full.adj.unrel05 \
--pheno ${workdir}input/PHQ9GAD7WSASworkimp_pheno_cov_batch_pcs_qcd_sexexcl_ibdexcl_MS_nohead.txt \
--qcovar ${workdir}input/age_agesq_pcs_covs.txt \
--covar ${workdir}input/sex_batch_covs.txt \
--reml-bivar-lrt-rg 0 \
--out ${workdir}results/gcta/phq9_rwsas4_greml_extraqc_full_adj_unrel05_covs_LRT0_rg \
--thread-num 10

EOT

sbatch ${workdir}scripts/BIGREML_rg_PHQ9WSAS4_sumstats.sh

```

## GAD7 WSAS-4
Sensitivity - compare to imputed WSAS and WSAS-5-c results
```{bash gad7 WSAS-4 bivariate greml, eval = F}

cd ~/credentials
workdir=$(sed '2q;d' paths.config.sh) 

rm ${workdir}scripts/BIGREML_rg_GAD7WSAS4_sumstats.sh
cat <<'EOT'>> ${workdir}scripts/BIGREML_rg_GAD7WSAS4_sumstats.sh
#!/bin/bash -l
#SBATCH --partition cpu
#SBATCH --time=0-48:00:00
#SBATCH --nodes=1
#SBATCH --ntasks=8
#SBATCH --mem-per-cpu=19G
#SBATCH --job-name=greml_rg_gad7rwsas4
#SBATCH --output=../output/greml_rg_gad7_rwsas4_covs_%j.out

workdir=$(sed '2q;d' paths.config.sh) 
gcta=$(sed '5q;d' paths.config.sh)

${gcta} \
--reml-bivar 2 4 \
--grm ${workdir}results/gcta/glad_gcta_extraqc_full.adj.unrel05 \
--pheno ${workdir}input/PHQ9GAD7WSASworkimp_pheno_cov_batch_pcs_qcd_sexexcl_ibdexcl_MS_nohead.txt \
--qcovar ${workdir}input/age_agesq_pcs_covs.txt \
--covar ${workdir}input/sex_batch_covs.txt \
--reml-bivar-lrt-rg 0 \
--out ${workdir}results/gcta/gad7_rwsas4_greml_extraqc_full_adj_unrel05_covs_LRT0_rg \
--thread-num 10

EOT

sbatch ${workdir}scripts/BIGREML_rg_GAD7WSAS4_sumstats.sh

```

## WSAS WSAS-4
Sensitivity/secondary - imputed WSAS rg with WSAS-4
```{bash wsas WSAS-4 bivariate greml, eval = F}

cd ~/credentials
workdir=$(sed '2q;d' paths.config.sh) 

rm ${workdir}scripts/BIGREML_rg_WSASWSAS4_sumstats.sh
cat <<'EOT'>> ${workdir}scripts/BIGREML_rg_WSASWSAS4_sumstats.sh
#!/bin/bash -l
#SBATCH --partition cpu
#SBATCH --time=0-48:00:00
#SBATCH --nodes=1
#SBATCH --ntasks=8
#SBATCH --mem-per-cpu=19G
#SBATCH --job-name=greml_rg
#SBATCH --output=../output/greml_rg_wsas_rwsas4_covs_%j.out

workdir=$(sed '2q;d' paths.config.sh) 
gcta=$(sed '5q;d' paths.config.sh)

${gcta} \
--reml-bivar 3 4 \
--grm ${workdir}results/gcta/glad_gcta_extraqc_full.adj.unrel05 \
--pheno ${workdir}input/PHQ9GAD7WSASworkimp_pheno_cov_batch_pcs_qcd_sexexcl_ibdexcl_MS_nohead.txt \
--qcovar ${workdir}input/age_agesq_pcs_covs.txt \
--covar ${workdir}input/sex_batch_covs.txt \
--reml-bivar-lrt-rg 0 \
--out ${workdir}results/gcta/wsas_rwsas4_greml_extraqc_full_adj_unrel05_covs_LRT0_rg \
--thread-num 10

EOT

sbatch ${workdir}scripts/BIGREML_rg_WSASWSAS4_sumstats.sh

```

## Investigate relationship between WSAS 4-item and Work item

### WSAS-4 Work
In all participants, rg between WSAS-4 and WSAS work item
```{bash WSAS-4 work bivariate greml, eval = F}

cd ~/credentials
workdir=$(sed '2q;d' paths.config.sh) 

rm ${workdir}scripts/BIGREML_rg_WSAS4work_sumstats.sh
cat <<'EOT'>> ${workdir}scripts/BIGREML_rg_WSAS4work_sumstats.sh
#!/bin/bash -l
#SBATCH --partition cpu
#SBATCH --time=0-48:00:00
#SBATCH --nodes=1
#SBATCH --ntasks=8
#SBATCH --mem-per-cpu=19G
#SBATCH --job-name=greml_rg
#SBATCH --output=../output/greml_rg_rwsas4_work_covs_%j.out

workdir=$(sed '2q;d' paths.config.sh) 
gcta=$(sed '5q;d' paths.config.sh)

${gcta} \
--reml-bivar 4 7 \
--grm ${workdir}results/gcta/glad_gcta_extraqc_full.adj.unrel05 \
--pheno ${workdir}input/PHQ9GAD7WSASworkimp_pheno_cov_batch_pcs_qcd_sexexcl_ibdexcl_MS_nohead.txt \
--qcovar ${workdir}input/age_agesq_pcs_covs.txt \
--covar ${workdir}input/sex_batch_covs.txt \
--reml-bivar-lrt-rg 0 \
--out ${workdir}results/gcta/rwsas4_work_greml_extraqc_full_adj_unrel05_covs_LRT0_rg \
--thread-num 10

EOT

sbatch ${workdir}scripts/BIGREML_rg_WSAS4work_sumstats.sh

```

### Completers' WSAS-4 work
In participants with data observed for all 5 items, rg between WSAS-4 and WSAS work item
```{bash compl WSAS-4 work bivariate greml, eval = F}

cd ~/credentials
workdir=$(sed '2q;d' paths.config.sh) 

rm ${workdir}scripts/BIGREML_rg_complWSAS4work_sumstats.sh
cat <<'EOT'>> ${workdir}scripts/BIGREML_rg_complWSAS4work_sumstats.sh
#!/bin/bash -l
#SBATCH --partition cpu
#SBATCH --time=0-48:00:00
#SBATCH --nodes=1
#SBATCH --ntasks=8
#SBATCH --mem-per-cpu=19G
#SBATCH --job-name=greml_rg
#SBATCH --output=../output/greml_rg_complrwsas4_work_covs_%j.out

workdir=$(sed '2q;d' paths.config.sh) 
gcta=$(sed '5q;d' paths.config.sh)

${gcta} \
--reml-bivar 5 7 \
--grm ${workdir}results/gcta/glad_gcta_extraqc_full.adj.unrel05 \
--pheno ${workdir}input/PHQ9GAD7WSASworkimp_pheno_cov_batch_pcs_qcd_sexexcl_ibdexcl_MS_nohead.txt \
--qcovar ${workdir}input/age_agesq_pcs_covs.txt \
--covar ${workdir}input/sex_batch_covs.txt \
--reml-bivar-lrt-rg 0 \
--out ${workdir}results/gcta/complrwsas4_work_greml_extraqc_full_adj_unrel05_covs_LRT0_rg \
--thread-num 10

EOT

sbatch ${workdir}scripts/BIGREML_rg_complWSAS4work_sumstats.sh

```

## Tables & plots with WSAS-5-c and WSAS-4

```{r Input table greml rg WSAS-5-c WSAS-4}

#rg
PHQ9.rg = c(1, 0.864929, 0.838224, 0.897794)                                    
GAD7.rg = c(0.864929, 1, 0.726371, 0.776238)        
WSAS.rg = c(0.838224, 0.726371, 1, 0.996710)
rWSAS4.rg = c(0.897794, 0.776238, 0.996710, 1)
greml.rg = rbind(PHQ9.rg, GAD7.rg, WSAS.rg, rWSAS4.rg)
colnames(greml.rg) = c("PHQ9", "GAD7", "WSAS-5-c", "WSAS-4")
rownames(greml.rg) = c("PHQ9", "GAD7", "WSAS-5-c", "WSAS-4")
greml.rg%>%kable(caption = "rgs",
                   digits = c(rep(2, 4)))%>%
  kable_styling()

#SE
PHQ9.rg.se = c(0, 0.057380, 0.088854, 0.087398)
GAD7.rg.se = c(0.057380, 0, 0.115507, 0.118468)
WSAS.rg.se = c(0.088854, 0.115507, 0, 0.005972)
rWSAS4.rg.se = c(0.087398, 0.118468, 0.005972, 0)
greml.rg.se = rbind(PHQ9.rg.se, GAD7.rg.se, WSAS.rg.se, rWSAS4.rg.se)
colnames(greml.rg.se) = c("PHQ9", "GAD7", "WSAS-5-c", "WSAS-4")
greml.rg.se%>%kable(caption = "rg SEs",
                   digits = c(rep(2, 4)))%>%
  kable_styling()

#95% CI
PHQ9.rg.lci = round(PHQ9.rg - (1.959964*PHQ9.rg.se),2)
PHQ9.rg.uci = round(PHQ9.rg + (1.959964*PHQ9.rg.se),2)
GAD7.rg.lci = round(GAD7.rg - (1.959964*GAD7.rg.se),2)
GAD7.rg.uci = round(GAD7.rg + (1.959964*GAD7.rg.se),2)
WSAS.rg.lci = round(WSAS.rg - (1.959964*WSAS.rg.se),2)
WSAS.rg.uci = round(WSAS.rg + (1.959964*WSAS.rg.se),2)
rWSAS4.rg.lci = round(rWSAS4.rg - (1.959964*rWSAS4.rg.se),2)
rWSAS4.rg.uci = round(rWSAS4.rg + (1.959964*rWSAS4.rg.se),2)
PHQ9.rg.ci = paste0(PHQ9.rg.lci, " - ", PHQ9.rg.uci)
GAD7.rg.ci = paste0(GAD7.rg.lci, " - ", GAD7.rg.uci)
WSAS.rg.ci = paste0(WSAS.rg.lci, " - ", WSAS.rg.uci)
rWSAS4.rg.ci = paste0(rWSAS4.rg.lci, " - ", rWSAS4.rg.uci)
greml.rg.ci = cbind(PHQ9.rg.ci, GAD7.rg.ci, WSAS.rg.ci, rWSAS4.rg.ci)
rownames(greml.rg.ci) = c("PHQ9", "GAD7", "WSAS-5-c", "WSAS-4")
colnames(greml.rg.ci) = c("PHQ9", "GAD7", "WSAS-5-c", "WSAS-4")
greml.rg.ci%>%
  kable(caption = "rg CIs")%>%
  kable_styling()

```

```{r Plot greml rg WSAS-5-c WSAS-4}

greml.rg = greml.rg%>%
  as.data.frame(.)%>%
  rownames_to_column(var = "measure1")%>%
  pivot_longer(cols = c(PHQ9, GAD7, `WSAS-5-c`, `WSAS-4`), 
               names_to = "measure2", 
               values_to = "value")%>%
  mutate(across(where(is.numeric), round, 3))%>%
  filter(!is.na(value))

greml.rg = greml.rg %>%
  mutate(measure1 = factor(measure1, levels = c("PHQ9", "GAD7", "WSAS-5-c", "WSAS-4")),
         measure2 = factor(measure2, levels = c("PHQ9", "GAD7", "WSAS-5-c", "WSAS-4")))

greml.rg%>%
  arrange(measure1, measure2) %>%
  group_by(measure1) %>%
  filter(row_number() >= which(measure1 == measure2)) %>%
ggplot(., 
       aes(measure2, measure1, fill = value))+
  geom_tile(color = "white")+
  scale_fill_gradient2(low = "blue",
                       high = "red",
                       mid = "white",
                       midpoint = 0,
                       limit = c(-1,1),
                       space = "Lab",
                       name="Pearson\nCorrelation") +
  coord_fixed() + 
geom_text(aes(measure2, measure1, label = value), 
          color = "black",
          size = 5) +
  labs(title = "Bivariate-GREML estimated rg") +
theme(axis.title = element_blank(),
      axis.text = element_text(colour="black",
                               size = 12,
                               angle = 30),
  panel.grid.major = element_blank(),
  panel.border = element_blank(),
  panel.background = element_blank(),
  axis.ticks = element_blank(),
  legend.justification = c(1, 0),
  legend.position = c(0.45, 0.7),
  legend.direction = "horizontal") +
  guides(fill = guide_colorbar(barwidth = 7,
                               barheight = 1,
                title.position = "top",
                title.hjust = 0.5))

```

Alternative plot: point with error bars
```{r data prep gcta rg point error bar WSAS-5-c WSAS-4}

measure1 <- factor(rep(c("PHQ9", "GAD7", "WSAS-5-c","WSAS-4"), each = 4, times = 2))
measure2 <- factor(rep(c("PHQ9", "GAD7", "WSAS-5-c", "WSAS-4"), 8))
corr.type <- factor(rep(c("genetic", "phenotypic"), each = 16))
rg <- c(1, 0.864929, 0.838224, 0.897794,
        0.864929, 1, 0.726371, 0.776238,
        0.838224, 0.726371, 1, 0.996710,
        0.897794, 0.776238, 0.996710, 1)
rg.se <- c(0, 0.057380, 0.088854, 0.087398, 
           0.057380, 0, 0.115507, 0.118468, 
           0.088854, 0.115507, 0, 0.005972, 
           0.087398, 0.118468, 0.005972, 0)
rg.lci <- rg-(1.959964*rg.se)
rg.uci <- rg+(1.959964*rg.se)
#rg.p
rgp <- c(rg, 
         1, 0.693, 0.629, 0.624,
         0.693, 1, 0.492, 0.486,
         0.629, 0.492, 1, 0.980,
         0.624, 0.486, 0.980, 1)
rgp.lci <- c(rg.lci, 
             1, 0.685, 0.620, 0.615,
             0.685, 1, 0.480, 0.475,
             0.620, 0.480, 1, 0.979,
             0.615, 0.475, 0.979, 1)
rgp.uci <- c(rg.uci,
             1, 0.701, 0.639, 0.633,
             0.701, 1, 0.504, 0.497, 
             0.639, 0.504, 1, 0.980, 
             0.633, 0.497, 0.980, 1)
#rgp.p <- c(rg.p, rep(0, length(rp)))
#rgp.sig <- rgp.p<(0.05/3)
rgp.plot.point <- data.frame(measure1, measure2, corr.type, rgp, rgp.lci, rgp.uci)
rgp.plot.point <- rgp.plot.point%>%
  mutate(measure1 = fct_relevel(measure1, "PHQ9", "GAD7", "WSAS-5-c", "WSAS-4"),
         measure2 = fct_relevel(measure2, "PHQ9", "GAD7", "WSAS-5-c", "WSAS-4"))%>%
   unite("corrmeasure", measure2, corr.type, sep = " ", remove = FALSE)%>%
  mutate(corrmeasure = fct_relevel(corrmeasure,
                                   c("PHQ9 genetic",
                                     "PHQ9 phenotypic",
                                     "GAD7 genetic",
                                     "GAD7 phenotypic",
                                     "WSAS-5-c genetic ",
                                     "WSAS-5-c phenotypic ",
                                     "WSAS-4 genetic",
                                     "WSAS-4 phenotypic")))

```

```{r Plot gcta rg point error bar WSAS-5-c WSAS-4}

ggplot(rgp.plot.point,
       aes(x=measure1,
           y=rgp,
           group=corrmeasure, 
           colour=corrmeasure,
           shape=corrmeasure)) + 
 geom_point(position=position_dodge(width=.5),
            size=3) +
  geom_errorbar(aes(ymin=rgp.lci, ymax=rgp.uci),
       width=.1,
       position=position_dodge(0.5)) +
  scale_fill_manual(values = Dark2rep) +
  scale_color_manual(values = Dark2rep) +
  scale_shape_manual(values = c(15,0,16,1,17,2,18,5)) + #hollow/filled square = 0/15, hollow/filled circle = 1/16, hollow/filled triangle = 2/17, hollow/filled diamond = 5/18
  scale_y_continuous(limits=c(0,1.02),
                     breaks=c(0,0.25,0.5,0.75,1)) +
  geom_hline(yintercept=as.numeric(c(0)),
             linetype="dotted",
             colour="darkgray")+
  labs(x = "Trait",
       y="Bivariate GREML rg") + 
   guides(colour = guide_legend(nrow=2,byrow=TRUE)) + 
  my_theme +
  theme(legend.title = element_blank(),
        legend.position = "bottom",
        panel.grid.major.y = element_line(linetype = "dotted",
                                          colour = "gray"))
ggsave("gcta.genpheno.corr.ccwsas.rwsas4.png", width = 18, height = 13, units = "cm", dpi=800)

```
